HelpMyStreetFE/HelpMyStreetFE/Controllers/RequestHelpApiController.cs

```diff
@@ -34,23 +34,48 @@ public async Task<ActionResult<bool>> AcceptRequest([FromBody]UpdateJobRequest J
                 if (!ModelState.IsValid)
                     throw new Exception("Job ID has not been supplied");
 
-                var userId = int.Parse(HttpContext.User.FindFirst(ClaimTypes.NameIdentifier).Value);
-                int jobId = -1;
-                int.TryParse(Base64Utils.Base64Decode(Job.JobID), out jobId);
+                var userId = int.Parse(HttpContext.User.FindFirst(ClaimTypes.NameIdentifier).Value);                             
+                return await _requestService.UpdateJobStatusToInProgressAsync(DecodeJobID(Job.JobID), userId, userId);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError("an error occured accepting help", ex);
+                return StatusCode(500);
+            }
+        }
 
-                if (jobId == -1)
-                    throw new Exception("Could not decode Job ID: " + Job.JobID);                                
+        [Authorize]
+        [HttpPost("complete-request")]
+        public async Task<ActionResult<bool>> CompleteRequest([FromBody]UpdateJobRequest Job)
+        {
+            try
+            {
+                if (!ModelState.IsValid)
+                    throw new Exception("Job ID has not been supplied");
```

Comment:
I should have noticed in an earlier PR this but if the ModelState is not valid you should return a BadRequest (400) as opposed to throwing an error with will result in a 500

Reply:
noted & implemented